### DISCUSSION TITLE
FSharpChecker.GetProjectOptionsFromScript --out flag not supported by design?

### DISCUSSION THREAD

**Discussion by queil** [dotnet/fsharp#17696] [Q&A] [answered]
Hi,

I've been trying using the `--out` compiler flag in `FSharpChecker.GetProjectOptionsFromScript` (via `otherFlags`) so that it populates the `outputDirectory` constructor param in my custom dependency manager. However, it fails with: `commandLineArgs (0,1)-(0,1) parameter error Unrecognized option: '--out'. Use '--help' to learn about recognized command line options.` Is this by design or a bug?

**Comment by vzarytovskii** [comment, marked as answer]
From the top of my head - I would say it's by design, since checker is not expected to output any assemblies as well as it's not really applicable for scripts. Don't remember thought which subset of flags does it use.

**Reply by queil** [reply]
Ok, it makes sense. Would there be any other mechanism available to pass an additional context to the dependency manager (apart from the `ResolveDependencies` method parameters)? Essentially, I want the dependency manager use the same output directory for its temporary files as the main compilation (and the output directory may be overridden by the user so I cannot assume a static path).
